### PR TITLE
Add pylint action

### PR DESCRIPTION
**Pull request by everreau**
- I've tested this in a private repo only thing that might still need tweaking is the specific repo pulling but it should be close.

### CODE REVIEW

**Comment by everreau** [comment]
Failure is expected since I haven't merged the PR with all the code clean up yet.

File: .github/workflows/pylint.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.9"]
+    steps:
+    - name: Checkout Janeway
+      uses: actions/checkout@v4
+      with:
+        repository: eScholarship/janeway
+        ref: e73827dc9db693756a3fe448f7a00d9963b2f821
```
Comment (bibliotechy) [inline .github/workflows/pylint.yml:20]: Do we want to be building against master? The latest release, using something like https://github.com/marketplace/actions/get-latest-release ?

Comment (everreau) [reply .github/workflows/pylint.yml:20]: well, neither of these options would have us build against what we're currently running in prd. Right now it's pulling the branch we have deployed but we'll have to update it every release.  There are changes in master/latest releases that break tests.

Comment (bibliotechy) [reply .github/workflows/pylint.yml:20]: @everreau Ah, ok. Maybe it's the kind of thing where in the future we do a matrix build against multiple releases of Janeway so we have some advanced notice if something is going to break

**Review by bibliotechy — APPROVED** [review]

**Pull request merged**